packages/gestalt/src/Tile/Tile.js

```diff
@@ -0,0 +1,189 @@
+// @flow strict
+import { type Node, useEffect, useState } from 'react';
+import classnames from 'classnames';
+import Box from '../Box.js';
+import InternalCheckbox from '../Checkbox/InternalCheckbox.js';
+import Flex from '../Flex.js';
+import { ENTER } from '../keyCodes.js';
+import TapArea, { type OnTapType } from '../TapArea.js';
+import Tooltip from '../Tooltip.js';
+import useFocusVisible from '../useFocusVisible.js';
+import useInteractiveStates from '../utils/useInteractiveStates.js';
+import { type Indexable } from '../zIndex.js';
+import styles from './Tile.css';
+
+type TooltipProps = {|
+  accessibilityLabel?: string,
+  inline?: boolean,
+  idealDirection?: 'up' | 'right' | 'down' | 'left',
+  text: string,
+  zIndex?: Indexable,
+|};
+
+type TileChangeHandler = ({|
+  event:
+    | SyntheticMouseEvent<HTMLDivElement>
+    | SyntheticKeyboardEvent<HTMLDivElement>
+    | SyntheticMouseEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLAnchorElement>,
+  selected: boolean,
+  id?: string,
+|}) => void;
+
+type Props = {|
+  /**
+   * The content to be wrapped by tile.
+   */
+  children?: Node,
+  /**
+   * The background color of the tile as a hex. Don't forget about tokenization.
+   */
+  bgColor?: string,
+  /**
+   * The border color of the tile. Don't forget about tokenization.
+   */
+  borderColor?: string,
+  /**
+   * Indicates if TileData should be disabled. Disabled TileData is inactive and cannot be interacted with.
+   */
+  disabled?: boolean,
+  /**
+   * Id to identify the tile.
+   */
+  id?: string,
+  /**
+   * Indicate if tile is in a selected state.
+   */
+  selected?: boolean,
+  /**
+   * Handler if the item selection state is changed.
+   */
+  onChange?: TileChangeHandler,
+  /**
+   * Shows a visible checkbox when the tile is selected state. See when using in a [group](http://gestalt.pinterest.systems/web/tiledata#Group).
+   */
+  showCheckbox?: boolean,
+  /**
+   * Adds a Tooltip on hover/focus of the Tile. See the With Tooltip variant to learn more.
+   */
+  tooltip?: TooltipProps,
+|};
+
+function ShouldUseTooltip({
+  children,
+  disabled,
+  tooltip,
+}: {|
+  children: Node,
+  disabled?: boolean,
+  tooltip?: TooltipProps,
+|}) {
+  if (!tooltip || disabled) return children;
+  return (
+    <Tooltip
+      accessibilityLabel={tooltip.accessibilityLabel}
+      inline={tooltip.inline}
+      idealDirection={tooltip.idealDirection || 'up'}
+      text={tooltip.text}
+      zIndex={tooltip.zIndex}
+    >
+      {children}
+    </Tooltip>
+  );
+}
+
+/**
+ * Used Internally to wrap a component with a Tile View
+ */
+export default function Tile({
+  bgColor,
+  borderColor,
+  children,
+  disabled = false,
+  id = '',
+  onChange,
+  selected = false,
+  showCheckbox,
+  tooltip,
+}: Props): Node {
+  const { handleOnBlur, handleOnMouseEnter, handleOnMouseLeave, isHovered } =
+    useInteractiveStates();
+
+  const [isSelected, setIsSelected] = useState(selected);
```

Comment:
This is a code smell. Why are we duplicating selection logic in the component? In what scenario would need to manipulate selected state internally and not rely on `selected` and `onChange`?

Reply:
Can you address this question please? Why are we duplicating selected logic with state?

Reply:
The ideal state makes sense — but that would involve a lot more control that we don't currently have internally. For now this is something that will have to remain solved by devs externally (similar to pretty much all our interactions, really)